### ISSUE TITLE
Lint: warn when `dataclass` subclass `__post_init__` doesn't call `super().__post_init__()`

### ISSUE THREAD

**Issue by janosh** [astral-sh/ty#2932]
per suggestion from @AlexWaygood in https://github.com/astral-sh/ty/issues/2931#issuecomment-3980171890:

### proposed rule

when a `@dataclass` subclass defines `__post_init__` and a parent class also defines `__post_init__`, emit a diagnostic if the subclass doesn't call `super().__post_init__(...)` (or doesn't forward required `InitVars`).

### motivation

forgetting to call the parent `__post_init__` silently skips field normalization, validation, and invariant setup — often causing subtle runtime bugs rather than clear errors. this is especially common in codebases with deep `@dataclass` inheritance (e.g. [torch-sim](https://github.com/TorchSim/torch-sim) where `SimState` → `MDState` → `NVTNoseHooverState` etc. all rely on parent `__post_init__` to guarantee field invariants).

### example

```py
from dataclasses import dataclass

@dataclass
class Parent:
    x: int | None = None

    def __post_init__(self) -> None:
        if self.x is None:
            self.x = 0

@dataclass
class Child(Parent):
    y: int = 0

    def __post_init__(self) -> None:
        # diagnostic: Parent.__post_init__ is not called
        self.y += 1
```

### refs

- https://github.com/astral-sh/ty/issues/2931
- https://github.com/astral-sh/ty/issues/2830
